src/components/URLImportDialog/index.js

```diff
@@ -0,0 +1,245 @@
+import React, { useState } from "react";
+import { injectIntl } from "react-intl";
+
+import { useTheme, makeStyles } from "@material-ui/core/styles";
+import { Publish as PublishIcon, Close as CloseIcon } from "@material-ui/icons";
+
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    IconButton,
+    TextField,
+    useMediaQuery,
+} from "@material-ui/core";
+
+import {
+    build as buildID,
+    formatMessage as fmt,
+    withI18N,
+} from "@cyverse-de/ui-lib";
+
+import intlData from "./messages";
+import ids from "./ids";
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        flexGrow: 1,
+    },
+    close: {
+        color: theme.palette.blueGrey,
+    },
+    container: {
+        display: "flex",
+        alignItems: "center", // without this the import button balloons in size vertically.
+
+        // Align things vertically on mobile.
+        [theme.breakpoints.down("sm")]: {
+            justifyContent: "center",
+            flexDirection: "column",
+        },
+    },
+    urlField: {
+        width: "100%",
+        margin: "0 10px 0 0",
```

Comment:
I think you can use `theme.spacing(0, 2, 0, 0)` here (https://material-ui.com/customization/spacing/).

Reply:
I used `theme.spacing(0, 2, 0, 0)`, seems to look okay.